### PR TITLE
Add Number & UUID validators

### PR DESCRIPTION
**Pull request by indyone**
Hi all,

As mentioned in the issue #46, it would be nice if we had a few validators built-in Vex.
With this PR, I am adding the number & UUID validators for Vex, which originally I had created as a separate Elixir package in the repo [StoiximanServices/vex_validators](https://github.com/StoiximanServices/vex_validators).
I've updated the code, tests and documentation, so that it meets Vex's standards and style :)

Please do take a look and give me your feedback.

### CODE REVIEW

File: lib/vex/validators/uuid.ex [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        :ok -> {:halt, :ok}
+        _ -> {:cont, error}
+      end
+    end)
+  end
+
+  defp do_validate(value, :not_any) do
+    error = {:error, "must not be a valid UUID string"}
+
+    Enum.reduce_while(@uuid_formats, :ok, fn format, _ ->
```
Comment (Zeeker) [inline lib/vex/validators/uuid.ex]: What about calling `do_validate(value, :any)`, this would take care of doing the reduce and make the code more DRY.

**Comment by indyone** [comment]
@Zeeker I've replaced the `Enum.reduce_while` in `do_validate(value, :not_any)` with a `case` statement that calls the `do_validate(value, :any)`

**Comment by indyone** [comment]
Hi all! What do you think about this PR? Is there anything else to comment on?

**Comment by indyone** [comment]
Guys thanks for merging my branch :)

Do you think that is a good idea to "release" a new version and also update hex?

**Comment by Zeeker** [comment]
I sadly don't have access the hex package, @jfornoff can you take care of that?

**Comment by jfornoff** [comment]
Sure! Vex 0.8.0 out!

**Pull request merged**